eclipse-che/che-devfile-registry
# chore: don't generate theia resources; remove sidecars; add devfile.yaml to work with the registry

### What does this PR do?

- Don't generate devworkspaces with che-theia editor
- Remove dockerfiles that used to use in dev components
- Add devfile.yaml to develop che-devfile-registry on the dogfooding cluster 

### Screenshot/screencast of this PR

I've used a devfile.yaml provided by the current changes to test if the registry works. For this I prepared OS instance with Che, added build capabilities to the cluster and:
1. Build registry by executing **1. Build and Publish a Custom Devfile Registry** command: 
![screenshot-nimbusweb me-2023 03 30-15_52_46](https://user-images.githubusercontent.com/1271546/228863199-3f5884b4-07f8-4171-9f26-7e3147e8c1e0.png)

2. Executed a command to use resulted image: **2. Configure Che to use the Custom Devfile Registry**
![screenshot-nimbusweb me-2023 03 30-15_56_28](https://user-images.githubusercontent.com/1271546/228863249-28809ba0-f0c2-4c36-a1f3-8a2e45ea7e9e.png)

3. Checked che-devfile-registry pod:
![screenshot-nimbusweb me-2023 03 30-15_58_04](https://user-images.githubusercontent.com/1271546/228863287-6489d2ab-d32b-4b02-9d50-c3458fd0f007.png)

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/22091

### How to test this PR?

- Build che-devfile registry
- Check generated `htdocs/devfiles/stack/` folders, they should contain generated devworkspaces only for che-code and che-idea editors
 
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
